runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -75,42 +85,63 @@ private static void loadVmProperties() {
      * @param metrics Metric Registry
      * @return True if metricsReportingSetup() function has been called earlier
      */
-    public static boolean isMetricsReportingSetUp(MetricRegistry metrics) {
+    public static boolean isMetricsReportingSetUp(@NonNull MetricRegistry metrics) {
         return metrics.getNames().contains(mpTrigger);
     }
 
     /**
-     * Start metrics reporting via the Dropwizard 'CsvReporter' file writer.
-     * Reporting can be turned on and off via the properties file described
-     * in loadVmProperties()'s docs.  The report interval and report
-     * directory cannot be altered at runtime.
+     * Start metrics reporting via the Dropwizard 'Slf4jReporter' and
+     * 'JmxReporter'. Reporting can be turned on and off via the system
+     * properties described in loadVmProperties()'s docs.
+     * The report interval and report directory cannot be altered at runtime.
      *
      * @param metrics Metrics registry
      */
-    public static void metricsReportingSetup(MetricRegistry metrics) {
+    public static void metricsReportingSetup(@NonNull MetricRegistry metrics) {
         metrics.counter(mpTrigger);
         loadVmProperties();
 
-        if (metricsReportingEnabled) {
-            MetricFilter f = (name, metric) -> {
-                if (name.equals(mpTrigger)) {
-                    loadVmProperties();
-                    return false;
-                }
-                return metricsReportingEnabled;
-            };
-
-            Slf4jReporter reporter1 = Slf4jReporter.forRegistry(metrics)
-                    .convertRatesTo(TimeUnit.SECONDS)
-                    .convertDurationsTo(TimeUnit.MILLISECONDS)
-                    .outputTo(LoggerFactory.getLogger("org.corfudb.metricsdata"))
-                    .filter(f)
-                    .build();
-            reporter1.start(interval, TimeUnit.SECONDS);
+        if (metricsCollectionEnabled) {
+            setupSlf4jReporting(metrics);
+            setupJmxReporting(metrics);
         }
     }
 
-    public static void addCacheGauges(MetricRegistry metrics, String name, Cache cache) {
+    // If enabled, setup jmx reporting
+    private static void setupJmxReporting(MetricRegistry metrics) {
+        if (!metricsJmxReportingEnabled) return;
+
+        // This filters noisy addressSpace metrics to have a clean JMX reporting
+        MetricFilter addressSpaceFilter = (name, metric) ->
+                !name.contains(ADDRESS_SPACE_METRIC_PREFIX);
```

Comment:
Is it just the Address Space Metrics which needs to be filtered? Should this be configurable like the slf4j configurable filter?

Reply:
That is a valid point. I decided against parametrizing it for two reasons. First: currently, the mentioned filter, filters the noisiest metrics that was making the visual representations less useful. For now it suffices. Second: this patch has the scope of enabling the reporting of current metrics to MBeans for the urgent need being able to examine them live. There is an open requirement to weed out and restructure the entire collected metrics which will affect the mechanisms for filtering but stays outside the scope of this commit.